### PR TITLE
move to penzai V2 API and independent treescope package

### PR DESCRIPTION
**Pull request by felixzinn**
visualization of pytrees is now possible with the standalone treescope package. This PR adapts to this change.
Furthermore penzai moved to a new V2 API, so a minor change to the visualization had to be made to accomodate to this.

### CODE REVIEW

**Review by felixzinn — COMMENTED** [review]
a `__treescope_repr__` for evermore components has been implemented which removes penzai as a dependency. Visualization of evermore components is now done with the standalone package treescope.

File: src/evermore/visualization.py [hunk 1]
```diff
@@ -1,110 +1,122 @@
 from __future__ import annotations
 
 import dataclasses
-import threading
+from collections.abc import Callable
```
Comment (pfackeldey) [inline src/evermore/visualization.py:4]: This should be imported from `typing` not `collections.abc` -> `from typing import Callable`

File: src/evermore/visualization.py [hunk 2]
```diff
@@ -1,110 +1,122 @@
...
+        path,
+        subtree_renderer,
+        fields_or_attribute_names=fields,
+        attr_style_fn=evermore_attr_style_fn_for_fields(fields),
+    )
+
+    # get colors for the background of the tree node
+    def _treescope_color(node) -> str:
+        """Returns the color of the tree node."""
+        from treescope import formatting_util
```
Comment (pfackeldey) [inline src/evermore/visualization.py:59]: `from treescope import formatting_util` is already imported in the beginning of this file.

File: src/evermore/visualization.py [hunk 3]
```diff
@@ -1,110 +1,122 @@
...
+    dataclass_util,
+    formatting_util,
+    renderers,
+    rendering_parts,
 )
 
 
-def convert_tree_to_penzai(tree: PyTree) -> PyTree:
-    from functools import partial
+class SupportTreescope:
```
Comment (pfackeldey) [inline src/evermore/visualization.py:15]: `SupportTreescope` -> `SupportsTreescope`

**Review by pfackeldey — CHANGES_REQUESTED** [review]

File: src/evermore/visualization.py (hunk 1 above)
Comment (felixzinn) [reply src/evermore/visualization.py:4]: `from typing import Callable` is **UP035** (deprecated-import) and automatically fixed by `pixi run lint` to `from collections.abc import Callable

**Review by pfackeldey — APPROVED** [review]

**Comment by pfackeldey** [comment]
awesome work!! 🎉

**Pull request merged**